Reenable slow gradcheck and make it pass

soulitzer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #80514

Context: For a while slow gradcheck CI was skipping nearly all tests and this hid the fact that it should've been failing and timing out (10+h runtime for TestGradients). The CI configuration has since been fixed to correct this, revealing the test failures. This PR reenables slow gradcheck CI and makes it pass again.

This PR:
- makes slow and failing tests run in fast gradcheck mode only
- reduce the input size for slow gradcheck only for unary/binary ufuncs (alternatively, skip the test entirely)
- skip entire test files on slow gradcheck runner if they don't use gradcheck (test_ops, test_meta, test_decomp, test_ops_jit)
- reduces the input size for some ops

Follow ups:
1. Investigate slow mode failures https://github.com/pytorch/pytorch/issues/80411
2. See if we can re-enable slow gradcheck tests for some of the slow tests by reducing the sizes of their inputs

The following are failing in slow mode, they are now running in fast mode only.
```
test_fn_fwgrad_bwgrad___rmod___cuda_float64
test_fn_fwgrad_bwgrad_linalg_householder_product_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_float64
test_fn_fwgrad_bwgrad_linalg_matrix_power_cuda_complex128
test_fn_fwgrad_bwgrad_cat_cuda_complex128
test_fn_fwgrad_bwgrad_linalg_lu_factor_ex_cuda_float64
test_fn_fwgrad_bwgrad_copysign_cuda_float64
test_fn_fwgrad_bwgrad_cholesky_inverse_cuda_complex128
test_fn_fwgrad_bwgrad_float_power_cuda_complex128
test_fn_fwgrad_bwgrad_fmod_cuda_float64
test_fn_fwgrad_bwgrad_float_power_cuda_float64
test_fn_fwgrad_bwgrad_linalg_lu_cuda_float64
test_fn_fwgrad_bwgrad_remainder_cuda_float64
test_fn_fwgrad_bwgrad_repeat_cuda_complex128
test_fn_fwgrad_bwgrad_prod_cuda_complex128
test_fn_fwgrad_bwgrad_slice_scatter_cuda_float64
test_fn_fwgrad_bwgrad_tile_cuda_complex128
test_fn_fwgrad_bwgrad_pow_cuda_float64
test_fn_fwgrad_bwgrad_pow_cuda_complex128
test_fn_fwgrad_bwgrad_fft_*
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

Marks the OpInfos for the following ops that run slowly in slow gradcheck as `fast_gradcheck` only (the left column represents runtime in seconds):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```
```
0  630.13  test_fn_gradgrad_nn_functional_conv2d_cuda_complex128 
1  81.086  test_fn_gradgrad_linalg_solve_triangular_cuda_complex128 
2  71.332  test_fn_gradgrad_norm_cuda_complex128 
3  64.308  test_fn_gradgrad__masked_std_cuda_complex128 
4  59.519  test_fn_gradgrad_div_no_rounding_mode_cuda_complex128 
5  58.836  test_fn_gradgrad_nn_functional_adaptive_avg_pool3
```

Reduces the sizes of the inputs for:
- diff
- diag_embed

soulitzer committed:
Reenable slow gradcheck and run failing tests in fast_mode only

[ghstack-poisoned]

soulitzer committed:
Update on "[Testing only] Reenable slow gradcheck and run failing tests in fast_mode only"




[ghstack-poisoned]

soulitzer committed:
Update on "[Testing only] Reenable slow gradcheck and run failing tests in fast_mode only"




[ghstack-poisoned]

seemethere commented:
Adding ciflow/periodic so that the periodic jobs will get triggered

soulitzer committed:
Update on "Reenable slow gradcheck and run failing tests in fast_mode only"


Make sure the following tests (that are failing in slow mode) are being in run in fast mode only. Follow-up: They should be investigated https://github.com/pytorch/pytorch/issues/80411 to ensure that there is no incorrect gradients:
```
test_fn_fwgrad_bwgrad___rmod___cuda_float64
test_fn_fwgrad_bwgrad_linalg_householder_product_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_float64
test_fn_fwgrad_bwgrad_linalg_matrix_power_cuda_complex128
test_fn_fwgrad_bwgrad_cat_cuda_complex128
test_fn_fwgrad_bwgrad_linalg_lu_factor_ex_cuda_float64
test_fn_fwgrad_bwgrad_copysign_cuda_float64
test_fn_fwgrad_bwgrad_cholesky_inverse_cuda_complex128
test_fn_fwgrad_bwgrad_float_power_cuda_complex128
test_fn_fwgrad_bwgrad_fmod_cuda_float64
test_fn_fwgrad_bwgrad_float_power_cuda_float64
test_fn_fwgrad_bwgrad_linalg_lu_cuda_float64
test_fn_fwgrad_bwgrad_remainder_cuda_float64
test_fn_fwgrad_bwgrad_repeat_cuda_complex128
test_fn_fwgrad_bwgrad_prod_cuda_complex128
test_fn_fwgrad_bwgrad_slice_scatter_cuda_float64
test_fn_fwgrad_bwgrad_tile_cuda_complex128
test_fn_fwgrad_bwgrad_pow_cuda_float64
test_fn_fwgrad_bwgrad_pow_cuda_complex128
test_fn_fwgrad_bwgrad_fft_*
```
TODO: may want to move reenable slow gradcheck to later in the stack to avoid timing out / OOM issues

[ghstack-poisoned]

soulitzer commented on test/test_ops_gradients.py:
```diff
@@ -44,7 +45,9 @@ def is_inplace(variant):
         include_conjugated_inputs = op.test_conjugated_samples and dtype.is_complex
-        samples = op.sample_inputs(device, dtype, requires_grad=True, include_conjugated_inputs=include_conjugated_inputs)
+
+        samples = op.sample_inputs(device, dtype, requires_grad=True, include_conjugated_inputs=include_conjugated_inputs,
```
@mruberry FYI, I added this back (currently used in UnaryUfuncs and BinaryUfuncs sample inputs only) as the tests would probably time out otherwise. Happy to go with the alternative of skipping UnaryUfuncs and BinaryUfuncs entirely in slow gradcheck too.

soulitzer committed:
Update on "Reenable slow gradcheck and make it pass"


Context: For a while slow gradcheck CI was skipping nearly all tests and this hid the fact that it should've been failing and timing out (10+h runtime for TestGradients). The CI configuration has since been fixed to correct this, revealing the test failures. This PR reenables slow gradcheck CI and makes it pass again.

This PR:
- makes slow and failing tests run in fast gradcheck mode only
- reduce the input size for slow gradcheck only for unary/binary ufuncs (alternatively, skip the test entirely)
- skip entire test files on slow gradcheck runner if they don't use gradcheck (test_ops, test_meta, test_decomp, test_ops_jit)
- reduces the input size for some ops

Follow ups:
1. Investigate slow mode failures https://github.com/pytorch/pytorch/issues/80411
2. See if we can re-enable slow gradcheck tests for some of the slow tests by reducing the sizes of their inputs

The following are failing in slow mode, they are now running in fast mode only.
```
test_fn_fwgrad_bwgrad___rmod___cuda_float64
test_fn_fwgrad_bwgrad_linalg_householder_product_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_float64
test_fn_fwgrad_bwgrad_linalg_matrix_power_cuda_complex128
test_fn_fwgrad_bwgrad_cat_cuda_complex128
test_fn_fwgrad_bwgrad_linalg_lu_factor_ex_cuda_float64
test_fn_fwgrad_bwgrad_copysign_cuda_float64
test_fn_fwgrad_bwgrad_cholesky_inverse_cuda_complex128
test_fn_fwgrad_bwgrad_float_power_cuda_complex128
test_fn_fwgrad_bwgrad_fmod_cuda_float64
test_fn_fwgrad_bwgrad_float_power_cuda_float64
test_fn_fwgrad_bwgrad_linalg_lu_cuda_float64
test_fn_fwgrad_bwgrad_remainder_cuda_float64
test_fn_fwgrad_bwgrad_repeat_cuda_complex128
test_fn_fwgrad_bwgrad_prod_cuda_complex128
test_fn_fwgrad_bwgrad_slice_scatter_cuda_float64
test_fn_fwgrad_bwgrad_tile_cuda_complex128
test_fn_fwgrad_bwgrad_pow_cuda_float64
test_fn_fwgrad_bwgrad_pow_cuda_complex128
test_fn_fwgrad_bwgrad_fft_*
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

Marks the OpInfos for the following ops that run slowly in slow gradcheck as `fast_gradcheck` only (the left column represents runtime in seconds):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```
```
0  630.13  test_fn_gradgrad_nn_functional_conv2d_cuda_complex128 
1  81.086  test_fn_gradgrad_linalg_solve_triangular_cuda_complex128 
2  71.332  test_fn_gradgrad_norm_cuda_complex128 
3  64.308  test_fn_gradgrad__masked_std_cuda_complex128 
4  59.519  test_fn_gradgrad_div_no_rounding_mode_cuda_complex128 
5  58.836  test_fn_gradgrad_nn_functional_adaptive_avg_pool3
```

Reduces the sizes of the inputs for:
- diff
- diag_embed


[ghstack-poisoned]

albanD commented on test/test_ops_gradients.py:
```diff
@@ -44,7 +45,9 @@ def is_inplace(variant):
-        samples = op.sample_inputs(device, dtype, requires_grad=True, include_conjugated_inputs=include_conjugated_inputs)
+
+        samples = op.sample_inputs(device, dtype, requires_grad=True, include_conjugated_inputs=include_conjugated_inputs,
+                                   small_inputs_only=os.environ.get('PYTORCH_TEST_WITH_SLOW_GRADCHECK', "0") == "1")
```
nit: have a small utility function in testing _internal to avoid copy-pasted env variable name!

albanD commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -2677,14 +2681,16 @@ def sample_inputs_elementwise_unary(
     if not op_kwargs:
         op_kwargs = {}
 
+    _L = S if kwargs.get("small_inputs_only", False) else L
```
nit: move with the others above

soulitzer replied:
Not sure if possible - they are supposed to apply for different sample inputs functions

albanD reviewed (approved):
That's a lot of failing ones :(
Thanks for taking the time to fix this!

soulitzer committed:
Update on "Reenable slow gradcheck and make it pass"


Context: For a while slow gradcheck CI was skipping nearly all tests and this hid the fact that it should've been failing and timing out (10+h runtime for TestGradients). The CI configuration has since been fixed to correct this, revealing the test failures. This PR reenables slow gradcheck CI and makes it pass again.

This PR:
- makes slow and failing tests run in fast gradcheck mode only
- reduce the input size for slow gradcheck only for unary/binary ufuncs (alternatively, skip the test entirely)
- skip entire test files on slow gradcheck runner if they don't use gradcheck (test_ops, test_meta, test_decomp, test_ops_jit)
- reduces the input size for some ops

Follow ups:
1. Investigate slow mode failures https://github.com/pytorch/pytorch/issues/80411
2. See if we can re-enable slow gradcheck tests for some of the slow tests by reducing the sizes of their inputs

The following are failing in slow mode, they are now running in fast mode only.
```
test_fn_fwgrad_bwgrad___rmod___cuda_float64
test_fn_fwgrad_bwgrad_linalg_householder_product_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_float64
test_fn_fwgrad_bwgrad_linalg_matrix_power_cuda_complex128
test_fn_fwgrad_bwgrad_cat_cuda_complex128
test_fn_fwgrad_bwgrad_linalg_lu_factor_ex_cuda_float64
test_fn_fwgrad_bwgrad_copysign_cuda_float64
test_fn_fwgrad_bwgrad_cholesky_inverse_cuda_complex128
test_fn_fwgrad_bwgrad_float_power_cuda_complex128
test_fn_fwgrad_bwgrad_fmod_cuda_float64
test_fn_fwgrad_bwgrad_float_power_cuda_float64
test_fn_fwgrad_bwgrad_linalg_lu_cuda_float64
test_fn_fwgrad_bwgrad_remainder_cuda_float64
test_fn_fwgrad_bwgrad_repeat_cuda_complex128
test_fn_fwgrad_bwgrad_prod_cuda_complex128
test_fn_fwgrad_bwgrad_slice_scatter_cuda_float64
test_fn_fwgrad_bwgrad_tile_cuda_complex128
test_fn_fwgrad_bwgrad_pow_cuda_float64
test_fn_fwgrad_bwgrad_pow_cuda_complex128
test_fn_fwgrad_bwgrad_fft_*
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

Marks the OpInfos for the following ops that run slowly in slow gradcheck as `fast_gradcheck` only (the left column represents runtime in seconds):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```
```
0  630.13  test_fn_gradgrad_nn_functional_conv2d_cuda_complex128 
1  81.086  test_fn_gradgrad_linalg_solve_triangular_cuda_complex128 
2  71.332  test_fn_gradgrad_norm_cuda_complex128 
3  64.308  test_fn_gradgrad__masked_std_cuda_complex128 
4  59.519  test_fn_gradgrad_div_no_rounding_mode_cuda_complex128 
5  58.836  test_fn_gradgrad_nn_functional_adaptive_avg_pool3
```

Reduces the sizes of the inputs for:
- diff
- diag_embed


[ghstack-poisoned]

soulitzer committed:
Update on "Reenable slow gradcheck and make it pass"


Context: For a while slow gradcheck CI was skipping nearly all tests and this hid the fact that it should've been failing and timing out (10+h runtime for TestGradients). The CI configuration has since been fixed to correct this, revealing the test failures. This PR reenables slow gradcheck CI and makes it pass again.

This PR:
- makes slow and failing tests run in fast gradcheck mode only
- reduce the input size for slow gradcheck only for unary/binary ufuncs (alternatively, skip the test entirely)
- skip entire test files on slow gradcheck runner if they don't use gradcheck (test_ops, test_meta, test_decomp, test_ops_jit)
- reduces the input size for some ops

Follow ups:
1. Investigate slow mode failures https://github.com/pytorch/pytorch/issues/80411
2. See if we can re-enable slow gradcheck tests for some of the slow tests by reducing the sizes of their inputs

The following are failing in slow mode, they are now running in fast mode only.
```
test_fn_fwgrad_bwgrad___rmod___cuda_float64
test_fn_fwgrad_bwgrad_linalg_householder_product_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_float64
test_fn_fwgrad_bwgrad_linalg_matrix_power_cuda_complex128
test_fn_fwgrad_bwgrad_cat_cuda_complex128
test_fn_fwgrad_bwgrad_linalg_lu_factor_ex_cuda_float64
test_fn_fwgrad_bwgrad_copysign_cuda_float64
test_fn_fwgrad_bwgrad_cholesky_inverse_cuda_complex128
test_fn_fwgrad_bwgrad_float_power_cuda_complex128
test_fn_fwgrad_bwgrad_fmod_cuda_float64
test_fn_fwgrad_bwgrad_float_power_cuda_float64
test_fn_fwgrad_bwgrad_linalg_lu_cuda_float64
test_fn_fwgrad_bwgrad_remainder_cuda_float64
test_fn_fwgrad_bwgrad_repeat_cuda_complex128
test_fn_fwgrad_bwgrad_prod_cuda_complex128
test_fn_fwgrad_bwgrad_slice_scatter_cuda_float64
test_fn_fwgrad_bwgrad_tile_cuda_complex128
test_fn_fwgrad_bwgrad_pow_cuda_float64
test_fn_fwgrad_bwgrad_pow_cuda_complex128
test_fn_fwgrad_bwgrad_fft_*
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

Marks the OpInfos for the following ops that run slowly in slow gradcheck as `fast_gradcheck` only (the left column represents runtime in seconds):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```
```
0  630.13  test_fn_gradgrad_nn_functional_conv2d_cuda_complex128 
1  81.086  test_fn_gradgrad_linalg_solve_triangular_cuda_complex128 
2  71.332  test_fn_gradgrad_norm_cuda_complex128 
3  64.308  test_fn_gradgrad__masked_std_cuda_complex128 
4  59.519  test_fn_gradgrad_div_no_rounding_mode_cuda_complex128 
5  58.836  test_fn_gradgrad_nn_functional_adaptive_avg_pool3
```

Reduces the sizes of the inputs for:
- diff
- diag_embed


[ghstack-poisoned]

soulitzer committed:
Update on "Reenable slow gradcheck and make it pass"


Context: For a while slow gradcheck CI was skipping nearly all tests and this hid the fact that it should've been failing and timing out (10+h runtime for TestGradients). The CI configuration has since been fixed to correct this, revealing the test failures. This PR reenables slow gradcheck CI and makes it pass again.

This PR:
- makes slow and failing tests run in fast gradcheck mode only
- reduce the input size for slow gradcheck only for unary/binary ufuncs (alternatively, skip the test entirely)
- skip entire test files on slow gradcheck runner if they don't use gradcheck (test_ops, test_meta, test_decomp, test_ops_jit)
- reduces the input size for some ops

Follow ups:
1. Investigate slow mode failures https://github.com/pytorch/pytorch/issues/80411
2. See if we can re-enable slow gradcheck tests for some of the slow tests by reducing the sizes of their inputs

The following are failing in slow mode, they are now running in fast mode only.
```
test_fn_fwgrad_bwgrad___rmod___cuda_float64
test_fn_fwgrad_bwgrad_linalg_householder_product_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_float64
test_fn_fwgrad_bwgrad_linalg_matrix_power_cuda_complex128
test_fn_fwgrad_bwgrad_cat_cuda_complex128
test_fn_fwgrad_bwgrad_linalg_lu_factor_ex_cuda_float64
test_fn_fwgrad_bwgrad_copysign_cuda_float64
test_fn_fwgrad_bwgrad_cholesky_inverse_cuda_complex128
test_fn_fwgrad_bwgrad_float_power_cuda_complex128
test_fn_fwgrad_bwgrad_fmod_cuda_float64
test_fn_fwgrad_bwgrad_float_power_cuda_float64
test_fn_fwgrad_bwgrad_linalg_lu_cuda_float64
test_fn_fwgrad_bwgrad_remainder_cuda_float64
test_fn_fwgrad_bwgrad_repeat_cuda_complex128
test_fn_fwgrad_bwgrad_prod_cuda_complex128
test_fn_fwgrad_bwgrad_slice_scatter_cuda_float64
test_fn_fwgrad_bwgrad_tile_cuda_complex128
test_fn_fwgrad_bwgrad_pow_cuda_float64
test_fn_fwgrad_bwgrad_pow_cuda_complex128
test_fn_fwgrad_bwgrad_fft_*
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

Marks the OpInfos for the following ops that run slowly in slow gradcheck as `fast_gradcheck` only (the left column represents runtime in seconds):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```
```
0  630.13  test_fn_gradgrad_nn_functional_conv2d_cuda_complex128 
1  81.086  test_fn_gradgrad_linalg_solve_triangular_cuda_complex128 
2  71.332  test_fn_gradgrad_norm_cuda_complex128 
3  64.308  test_fn_gradgrad__masked_std_cuda_complex128 
4  59.519  test_fn_gradgrad_div_no_rounding_mode_cuda_complex128 
5  58.836  test_fn_gradgrad_nn_functional_adaptive_avg_pool3
```

Reduces the sizes of the inputs for:
- diff
- diag_embed


[ghstack-poisoned]

soulitzer committed:
Update on "Reenable slow gradcheck and make it pass"


Context: For a while slow gradcheck CI was skipping nearly all tests and this hid the fact that it should've been failing and timing out (10+h runtime for TestGradients). The CI configuration has since been fixed to correct this, revealing the test failures. This PR reenables slow gradcheck CI and makes it pass again.

This PR:
- makes slow and failing tests run in fast gradcheck mode only
- reduce the input size for slow gradcheck only for unary/binary ufuncs (alternatively, skip the test entirely)
- skip entire test files on slow gradcheck runner if they don't use gradcheck (test_ops, test_meta, test_decomp, test_ops_jit)
- reduces the input size for some ops

Follow ups:
1. Investigate slow mode failures https://github.com/pytorch/pytorch/issues/80411
2. See if we can re-enable slow gradcheck tests for some of the slow tests by reducing the sizes of their inputs

The following are failing in slow mode, they are now running in fast mode only.
```
test_fn_fwgrad_bwgrad___rmod___cuda_float64
test_fn_fwgrad_bwgrad_linalg_householder_product_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_complex128
test_fn_fwgrad_bwgrad__masked_prod_cuda_float64
test_fn_fwgrad_bwgrad_linalg_matrix_power_cuda_complex128
test_fn_fwgrad_bwgrad_cat_cuda_complex128
test_fn_fwgrad_bwgrad_linalg_lu_factor_ex_cuda_float64
test_fn_fwgrad_bwgrad_copysign_cuda_float64
test_fn_fwgrad_bwgrad_cholesky_inverse_cuda_complex128
test_fn_fwgrad_bwgrad_float_power_cuda_complex128
test_fn_fwgrad_bwgrad_fmod_cuda_float64
test_fn_fwgrad_bwgrad_float_power_cuda_float64
test_fn_fwgrad_bwgrad_linalg_lu_cuda_float64
test_fn_fwgrad_bwgrad_remainder_cuda_float64
test_fn_fwgrad_bwgrad_repeat_cuda_complex128
test_fn_fwgrad_bwgrad_prod_cuda_complex128
test_fn_fwgrad_bwgrad_slice_scatter_cuda_float64
test_fn_fwgrad_bwgrad_tile_cuda_complex128
test_fn_fwgrad_bwgrad_pow_cuda_float64
test_fn_fwgrad_bwgrad_pow_cuda_complex128
test_fn_fwgrad_bwgrad_fft_*
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

Marks the OpInfos for the following ops that run slowly in slow gradcheck as `fast_gradcheck` only (the left column represents runtime in seconds):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```
```
0  630.13  test_fn_gradgrad_nn_functional_conv2d_cuda_complex128 
1  81.086  test_fn_gradgrad_linalg_solve_triangular_cuda_complex128 
2  71.332  test_fn_gradgrad_norm_cuda_complex128 
3  64.308  test_fn_gradgrad__masked_std_cuda_complex128 
4  59.519  test_fn_gradgrad_div_no_rounding_mode_cuda_complex128 
5  58.836  test_fn_gradgrad_nn_functional_adaptive_avg_pool3
```

Reduces the sizes of the inputs for:
- diff
- diag_embed


[ghstack-poisoned]

soulitzer commented:
@pytorchbot merge -g
